src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -56,10 +56,17 @@ private partial class CompilationTracker
                 public readonly bool DocumentsAreFinalAndFrozen;
 
                 public CompilationTrackerGeneratorInfo(
+                    TextDocumentStates<SourceGeneratedDocumentState> documents,
+                    GeneratorDriver? driver,
+                    bool documentsAreFinal) : this(documents, driver, documentsAreFinal, documentsAreFinalAndFrozen: false)
+                {
+                }
```

Comment:
@jasonmalinowski this made it easier to reason about external callers, knowing they could never pass a value for this.  Only the withers in this method can call the private constructor.

Reply:
Sounds reasonable to me!